### PR TITLE
increase visibility of video duration in thumbnails

### LINKED ISSUE #7: Visibility of video duration in thumbnail
**Checklist**
- [x] I made sure that there are no existing issues - open or closed - to which I could contribute my information.
- [x]  I have read the FAQ and my problem isn't listed.
- [x]  I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [x]  This issue contains only one feature request.
- [ ]  I have read and understood the [contribution guidelines.](https://github.com/FossifyX/General-Discussion?tab=readme-ov-file#contribution-rules-for-developers)

**Is your feature request related to a problem? Please describe.**
The video duration shown in thumbnails can sometimes be hard to read depending on the thumbnail colors.

**Describe the solution you'd like**
I would recommend something like YouTube thumbnails where they put the video duration in a semi-transparent black box for better visibility.
thumbnail

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by gmasil**
Resolves #7 

**TLDR:** Duration of videos in thumbnails has bad visibility on light backgrounds. This PR ~~adds a half-transperent background with rounded corners to the video duration text~~ combines the play icon and the video duration text (if enabled).

Screenshot of result is posted ~~there~~ below.

### CODE REVIEW

**Comment by naveensingh** [comment]
Why not merge both of the elements? A single rectangular background with the play icon followed by the video duration. It'll look better. I think Samsung's gallery has it like that.

You can experiment with the corner radius and the height to see what looks good or I'll do it before merging the PR.

**Comment by Aga-C** [comment]
If you change `Related issue: #7` to `Closes #7`, it will close the original issue automatically. It helps to manage what has already been fixed.

More info: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

**Comment by gmasil** [comment]
Thanks for pointing out the linking of issues!

@naveensingh what do you think about this? Did I understand you correctly?

**Comment by naveensingh** [comment]
The screenshot looks good but the text doesn't look like it's aligned with the icon horizontally. I'll be back later.

**Comment by gmasil** [comment]
I gave it a shot:

**Comment by jcperil** [comment]
Hi, what do you think about the pill shaped like below?

<p align="center"></p>

Implemented in VLC Media Player

<p align="center"></p>

<p align="center"></p>

**Comment by naveensingh** [comment]
@jcperil that's how it's usually done but the duration text view can be toggled off and then the play icon alone won't look good with a rectangular background.

**Review by naveensingh — CHANGES_REQUESTED** [review]

File: app/src/main/res/drawable/darkened_automatic_circle_background.xml
Comment (gmasil) [reply app/src/main/res/drawable/darkened_automatic_circle_background.xml:3]: That's the idea, it automatically forms a circle. That works for everything smaller than the given radius. I can change it to the `selection_check_size` and it will work just fine for this scenario, but not generically anymore. Your call.

File: app/src/main/res/layout/video_item_grid.xml
Comment (gmasil) [reply app/src/main/res/layout/video_item_grid.xml:53]: Yes, it should work, but it is slightly misaligned without that extra padding. These 2dp extra padding is the difference between my two screenshots, in the first you spotted the slight misalignment. Sadly I am not able to fix the misalignment without this little hack, I am new to android layouts and not a frontend dev in general.
* I can remove it, for this slight misalignment
* I can leave it in
* You try to find a fix that I couldn't find

Comment (gmasil) [reply app/src/main/res/layout/video_item_grid.xml:53]: Now I am not sure if I can see straight... I removed the extra padding and I think it looks correct.

File: app/src/main/res/drawable/darkened_automatic_circle_background.xml
Comment (naveensingh) [reply app/src/main/res/drawable/darkened_automatic_circle_background.xml]: it looks fine without the padding :)

File: app/src/main/res/layout/video_item_grid.xml
Comment (naveensingh) [reply app/src/main/res/layout/video_item_grid.xml:53]: it looks fine without the padding :)

File: app/src/main/res/drawable/darkened_automatic_circle_background.xml
Comment (naveensingh) [reply app/src/main/res/drawable/darkened_automatic_circle_background.xml]: `selection_check_size` is okay

**Comment by naveensingh** [comment]
thanks :)

**Comment by naveensingh** [comment]
@gmasil I'll check those other PRs sometime later, right now focusing on the release.

**Comment by gmasil** [comment]
@naveensingh thanks, I am sure you have enough on your plate with the release, there is no hurry :)

**Pull request merged**
